### PR TITLE
Add Slint

### PR DESCRIPTION
**Pull request by ogoffart**
## Description

Add the [Slint](https://slint.dev) language

## Checklist:

- [X] **I am adding a new language.**
  - [X] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      -  https://github.com/search?q=path%3A*.slint+-is%3Afork+-repo%3Aslint-ui%2Fslint&type=code
  - [X] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - Coming from the examples in https://github.com/slint-ui/slint
    - Sample license(s): MIT
  - [X] I have included a syntax highlighting grammar: https://github.com/slint-ui/slint-tmLanguage
      
  - [X] I have added a color
    - Hex value: `#2379F4`
    - Rationale: Color mentioned there: https://github.com/slint-ui/slint/tree/master/logo#color
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.
    - I don't think there are other languages with the same extension

### CODE REVIEW

**Review by lildude — CHANGES_REQUESTED** [review]
Two of your samples are too big as they're being suppressed by the diff by default. Please replace these with smaller samples.

You need to also add the cached license `.yml` file that would have been created when you ran `script/add-grammar` to add the grammar.

**Comment by ogoffart** [comment]
> Two of your samples are too big as they're being suppressed by the diff by default. Please replace these with smaller samples.

I edited them to removed big parts so they are small enough.

> You need to also add the cached license .yml file that would have been created when you ran script/add-grammar to add the grammar.

Added.

File: vendor/licenses/git_submodule/slint-tmLanguage.dep.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+name: slint-tmLanguage
+version: 1ad143138d60211dde6af6d9c2ad2a2d0d626c92
+type: git_submodule
+homepage: https://github.com/slint-ui/slint-tmLanguage.git
+license: none
+licenses: []
+notices: []
```
Comment (lildude) [inline vendor/licenses/git_submodule/slint-tmLanguage.dep.yml:8]: This should be populated with the deets of the license from this repo.

Try manually running this command https://github.com/github-linguist/linguist/blob/ca188d5b069684d5160170e2ab86fa995ca7fe0f/script/add-grammar#L134

**Comment by ogoffart** [comment]
I had forgotten to `git add` this file. I tried to clean everything and re-create it and that didn't work at first. Now this should be the correct one.

**Comment by lildude** [comment]
> I had forgotten to `git add` this file. I tried to clean everything and re-create it and that didn't work at first. Now this should be the correct one.

Whoops! There's only one file change now.

**Comment by ogoffart** [comment]
> Whoops! There's only one file change now.

Sorry about that. 
I'm doing this in codespaces which I'm not familiar with. Should be alright now.

**Review by lildude — APPROVED** [review]
All looks good now.

**Pull request merged**